crates/re_space_view_tensor/src/space_view_class.rs

```diff
@@ -430,10 +430,11 @@ fn colormap_preview_ui(
     re_tracing::profile_function!();
 
     let desired_size = egui::vec2(128.0, 16.0);
-    let (rect, response) = ui.allocate_exact_size(desired_size, egui::Sense::hover());
+    let (response, painter) = ui.allocate_painter(desired_size, egui::Sense::hover());
```

Comment:
This creates a `painter` with a smaller `clip_rect`, so that is probably the change that made it work.

This means `render_image` still has a footgun: an undocumented requirement that the clip rectangle is the same as the viewport rectangle.

This is a new requirement (it obviously wasn't there when the colormap gradient code was first written) so it would probably be good to understand it a bit more.

I am curious what happens, for instance, if the painter clip rectangle is _smaller_ than the viewport (e.g. when scrolling an area where we call `render_image`). It seems likely that will also be buggy.

Reply:
spot on, the **actual** error was a very simple mistake in the re_renderer callback!